lib/Dialect/FIRRTL/Transforms/InjectDUTHierarchy.cpp

```diff
@@ -145,6 +184,15 @@ void InjectDUTHierarchy::runOnOperation() {
     AnnotationSet::removeAnnotations(wrapper, [](auto) { return true; });
   }
 
+  // Compute a set of paths that are used _inside_ the wrapper.
+  DenseSet<StringAttr> wrapperComponentPaths;
+  wrapper.getBody()->walk([&](Operation *op) {
+    for (auto anno : AnnotationSet(op)) {
+      if (auto sym = anno.getMember<FlatSymbolRefAttr>("circt.nonlocal"))
+        wrapperComponentPaths.insert(sym.getAttr());
+    }
+  });
+
```

Comment:
It might be better to track the NLAs referenced by the DUT module annotations instead of the ones in the body of the module, especially since we don't have to walk the entire body of the module. 

I think MemOps and InstanceOps can have PortAnnotations which isn't handled here.

Reply:
Great idea! This saves a lot of work. The current patch uses this approach. Now, a clone is created if the DUT or its ports use an NLA. The internal components in the wrapper will use the original symbol. There is no checking for uses of the symbol by internal components meaning that this may produce a dead path.